molecule/release-upgrade/molecule.yml

```diff
@@ -1,67 +1,141 @@
 ---
-dependency:
-  name: galaxy
-  options:
-    role-file: requirements.yml
-driver:
-  name: docker
-lint: |
-    yamllint .
-    ansible-lint
-    # rubocop
-# This is ignored by molecule, but can be reused in yaml
-.platform_base: &platform_base
-  privileged: False
-  volumes:
-    - /sys/fs/cgroup:/sys/fs/cgroup:ro
-  tmpfs:
-    # Fixes issues accessing .so files under /tmp during pip builds.
-    # https://github.com/docker/compose/issues/1339
-    - /tmp:exec,mode=1777
-    - /run
-    - /run/lock
-  capabilities:
-    - NET_ADMIN
-    - NET_RAW
-  pre_build_image: true
-platforms:
-  - <<: *platform_base
-    name: centos-7
-    # molecule often fails to pull, so we pull all images in .travis.yml
-    image: quay.io/pulp/pulp_rpm-ci-c7:3.1.0
-    command: /sbin/init
-  - <<: *platform_base
-    name: debian-11
-    image: quay.io/pulp/pulp-ci-dbullseye:3.14.4
-    command: /sbin/init
-  - <<: *platform_base
-    name: fedora-33
-    image: quay.io/pulp/pulp_rpm-ci-f33:3.9.0
-    command: /usr/sbin/init
-provisioner:
-  name: ansible
-  env:
-    ANSIBLE_COLLECTIONS_PATH: $PWD/build/collections  # for ansible >= 2.10.0.dev0
-    ANSIBLE_COLLECTIONS_PATHS: $PWD/build/collections
-  config_options:
-    defaults:
-      stdout_callback: debug
-      callback_enabled: profile_tasks
-  inventory:
-    links:
-      host_vars: ../scenario_resources/host_vars/
-scenario:
-  test_sequence:
-    - dependency
-    - lint
-    - syntax
-    - create
-    - prepare
-    - converge
-    - idempotence
-    - check
-    - side_effect
-    - verify
-verifier:
-  name: ansible
-  directory: tests
+- hosts: all
+  tasks:
+    - set_fact:
+       pulpcore_version: "3.14.5"
+    - set_fact:
+       pulp_install_plugins:
+         pulp_file:
+           version: "1.8.0"
+         pulp_rpm:
+           version: "3.13.0"
+      when: ansible_facts.distribution in ['RedHat', 'CentOS']
+    - set_fact:
+       pulp_install_plugins:
+         pulp_file:
+           version: "1.8.0"
+      when: ansible_facts.os_family == 'Debian'
+    - include_role:
+        name: pulp_all_services
+    - name: get pulpcore version
+      command: /usr/local/lib/pulp/bin/pip freeze
+      register: pip_pkgs
```

Comment:
We have a better way of doing this with a pip module.

See ./pulp_common/tasks/install_pip.yml